test/unit-test/mqtt_agent_utest.c

```diff
@@ -1299,128 +1330,197 @@ void test_MQTTAgent_CommandLoop_with_eventCallback( void )
 
     TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
     /* Ensure that callback is invoked. */
-    TEST_ASSERT_EQUAL( 8, commandCompleteCallbackCount );
+    TEST_ASSERT_EQUAL( 1, commandCompleteCallbackCount );
     /* Ensure that acknowledgment is cleared. */
     TEST_ASSERT_EQUAL( 0, mqttAgentContext.pPendingAcks[ 0 ].packetId );
     TEST_ASSERT_EQUAL( NULL, mqttAgentContext.pPendingAcks[ 0 ].pOriginalCommand );
 
     /* Invoking mqttEventCallback with MQTT_PACKET_TYPE_PUBLISH packet type. */
     packetType = MQTT_PACKET_TYPE_PUBLISH;
+    commandCompleteCallbackCount = 0;
 
     MQTTAgentCommand_Publish_ExpectAnyArgsAndReturn( MQTTSuccess );
-    MQTTAgentCommand_Publish_ReturnThruPtr_pReturnFlags( &pReturnFlags );
+    MQTTAgentCommand_Publish_ReturnThruPtr_pReturnFlags( &returnFlags );
     MQTT_ProcessLoop_Stub( MQTT_ProcessLoop_CustomStub );
 
     mqttStatus = MQTTAgent_CommandLoop( &mqttAgentContext );
 
     TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
     /* Ensure that callback is invoked. */
-    TEST_ASSERT_EQUAL( 9, commandCompleteCallbackCount );
+    TEST_ASSERT_EQUAL( 1, commandCompleteCallbackCount );
     TEST_ASSERT_EQUAL( 1, publishCallbackCount );
 
     /* Test case when completion callback is NULL. */
     commandToSend.pCommandCompleteCallback = NULL;
+    commandCompleteCallbackCount = 0;
 
     MQTTAgentCommand_Publish_ExpectAnyArgsAndReturn( MQTTSuccess );
-    MQTTAgentCommand_Publish_ReturnThruPtr_pReturnFlags( &pReturnFlags );
+    MQTTAgentCommand_Publish_ReturnThruPtr_pReturnFlags( &returnFlags );
     MQTT_ProcessLoop_Stub( MQTT_ProcessLoop_CustomStub );
 
     mqttStatus = MQTTAgent_CommandLoop( &mqttAgentContext );
 
     TEST_ASSERT_EQUAL( MQTTSuccess, mqttStatus );
     /* Ensure that callback is not invoked. */
-    TEST_ASSERT_EQUAL( 9, commandCompleteCallbackCount );
+    TEST_ASSERT_EQUAL( 0, commandCompleteCallbackCount );
+}
+
+/**
+ * @brief Test MQTTAgent_CommandLoop failure case via calling MQTT_ProcessLoop multiple times.
+ */
+void test_MQTTAgent_CommandLoop_ProcessLoop_failure( void )
+{
+    MQTTStatus_t mqttStatus;
+    MQTTAgentContext_t mqttAgentContext;
+
+    setupAgentContext( &mqttAgentContext );
+
+    mqttAgentContext.mqttContext.connectStatus = MQTTConnected;
+    returnFlags.addAcknowledgment = false;
+    returnFlags.runProcessLoop = true;
+    returnFlags.endLoop = false;
+
+    MQTTAgentCommand_ProcessLoop_ExpectAnyArgsAndReturn( MQTTSuccess );
+    MQTTAgentCommand_ProcessLoop_ReturnThruPtr_pReturnFlags( &returnFlags );
+    /* Calling MQTT_ProcessLoop multiple times and returning failure second time. */
+    MQTT_ProcessLoop_Stub( MQTT_ProcessLoop_stub_multiple );
+
+    mqttStatus = MQTTAgent_CommandLoop( &mqttAgentContext );
+
+    TEST_ASSERT_EQUAL( MQTTRecvFailed, mqttStatus );
+}
+
+/**
+ * @brief Test MQTTAgent_CommandLoop behavior processing multiple commands in command queue.
+ */
+void test_MQTTAgent_CommandLoop_with_multiple_commands( void )
```

Comment:
What is this test checking - 1. Second command fails to get posted to the queue. OR 2. Second command gets posted to the queue and fails while processing? Would you please update the test name and comment accordingly.

Reply:
It is testing the second scenario. I have updated the comment.